extension/encoding/awslogsencodingextension/extension.go

```diff
@@ -31,29 +41,31 @@ func newExtension(cfg *Config, settings extension.Settings) (*encodingExtension,
 		return &encodingExtension{
 			unmarshaler: subscriptionfilter.NewSubscriptionFilterUnmarshaler(settings.BuildInfo),
 			format:      formatCloudWatchLogsSubscriptionFilter,
+			gzipPool:    sync.Pool{},
 		}, nil
 	case formatVPCFlowLog:
-		unmarshaler, err := vpcflowlog.NewVPCFlowLogUnmarshaler(
+		unmarshaler := vpcflowlog.NewVPCFlowLogUnmarshaler(
 			cfg.VPCFlowLogConfig.FileFormat,
 			settings.BuildInfo,
 			settings.Logger,
 		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create encoding extension for %q format: %w", formatVPCFlowLog, err)
-		}
```

Comment:
Why did you move the error handling? IIUC this means that specifying an invalid/unsupported format will now become a runtime error, rather than on startup - I don't think that's desirable.

Reply:
Thank you, I have put it back